packages/flutter/test/material/snack_bar_test.dart

```diff
@@ -1146,4 +1053,283 @@ void main() {
     expect(find.text('hello'), findsOneWidget);
     expect(called, 1);
   });
+
+  group('SnackBar position', () {
+    for (final SnackBarBehavior behavior in SnackBarBehavior.values) {
+      final SnackBar snackBar = SnackBar(
+        content: const Text('SnackBar text'),
+        behavior: behavior,
+      );
+
+      testWidgets(
+        '$behavior should align SnackBar with the bottom of Scaffold '
+        'when Scaffold has no other elements',
+        (WidgetTester tester) async {
+          await tester.pumpWidget(
+            MaterialApp(
+              home: Scaffold(
+                body: Container(),
+              ),
+            ),
+          );
+
+          final ScaffoldState scaffoldState = tester.state(find.byType(Scaffold));
+          scaffoldState.showSnackBar(snackBar);
+
+          await tester.pump();
```

Comment:
You might want the SnackBar to fully animate out before checking the positions.
```suggestion
          await tester.pumpAndSettle(); // Have the SnackBar fully animate out.
```

Reply:
Fixed.